include/revng-c/mlir/Dialect/Clift/IR/CliftOps.td

```diff
@@ -26,6 +27,120 @@ def Clift_StatementRegion : Region<CPred<"::mlir::clift::impl::verifyRegionConte
 def Clift_ExpressionRegion : Region<CPred<"::mlir::clift::impl::verifyRegionContent<::mlir::clift::ExpressionOpInterface>($_self)">,
                                     "Region representing an expression">;
 
+//===------------------------ Top level operations ------------------------===//
+
+def Clift_ModuleOp : Clift_Op<"module",
+                              [SymbolTable,
+                               HasOnlyGraphRegion,
+                               NoRegionArguments,
+                               NoTerminator,
+                               SingleBlock,
+                               RegionKindInterface]> {
+  let regions = (region AnyRegion:$body);
+
+  let skipDefaultBuilders = 1;
+  let builders = [OpBuilder<(ins)>];
+
+  let assemblyFormat = [{
+    $body attr-dict
+  }];
+
+  let hasVerifier = 1;
+}
+
+def Clift_FunctionOp : Clift_Op<"func",
+                                [Symbol,
+                                 CallableOpInterface,
+                                 FunctionOpInterface,
```

Comment:
I see that you have done the homework on figuring out what of the methods defined below are part of these interfaces. Thanks!